mdsuite/utils/molecule.py

```diff
@@ -0,0 +1,67 @@
+"""
+MDSuite: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+Module for the MDSuite molecule dataclass
+"""
+from dataclasses import dataclass
+
+
+@dataclass
+class Molecule:
+    """
+    Data class to define a molecule.
+
+    Attributes
+    ----------
+    name : str
+            Name of the molecule. This name will be stored in the database.
+    smiles : str (optional)
+            SMILES string to use in the definition of the molecule internally.
+            e.g. CCN1C=C[N+](+C1)C
+    species_dict : dict (optional)
+            A species dict for a custom molecule in the case where a SMILES string
+            cannot be written.
+            e.g. {'C': 6, 'N': 2, 'H': 12}
+    amount : int
```

Comment:
In the future I would like to allow for a flexible amount argument I cases where bonding may occur in a system. Before this is possible we will need to provide strict graph isomorphism checks to be even more confident in the bonding structure. As for cutoff and reference configuration, I would actually like to remove them as arguments and add an optimizer to the code that finds an optimal solution. That would eliminate their need in general. One can use atomistic RDFs now to get a good idea of cutoff but it could be done automatically.